io/net/virt-net/lpm.py

```diff
@@ -56,10 +56,19 @@ def setUp(self):
         self.hmc_pwd = self.params.get("hmc_pwd", '*', default='********')
         self.options = self.params.get("options", default='')
         self.net_device_type = self.params.get("net_device_type", default='')
-        self.lpar = self.get_partition_name("Partition Name")
+
+        if not self.params.get("lpar", default=''):
+            self.lpar = self.get_partition_name("Partition Name")
+            self.log.info("'lpar' parameter empty, migrating current LPAR")
+        else:
+            self.lpar = self.params.get("lpar", default='')
         if not self.lpar:
             self.cancel("LPAR Name not got from lparstat command")
-        self.lpar_ip = self.get_mcp_component("MNName")
+        if not self.params.get("lpar_ip", default=''):
+            self.lpar_ip = self.get_mcp_component("MNName")
+            self.log.info("'lpar_ip' parameter empty, migrating current LPAR")
+        else:
+            self.lpar_ip = self.params.get("lpar_ip", default='')
```

Comment:
You're kinda using the same logic here. I would combine both if statements. if the file has these two parameters then do this else that.

Reply:
Additionally, I would create a new class variable here called  "self.using_peer" or whatever indicating that you are migrating from a peer lpar. This will help you complete the service code much easier.